### ISSUE TITLE
development

### ISSUE THREAD

**Issue by patience4711** [patience4711/ESP32-read-APS-inverters#4]
I finished the testing of this software and ended up with a new version. There where lots of problems to solve, including system crashes at certain string operations like strncat and strncpy. 
To gain more memory i improved some of the webpages. Furthermore the pairing process was fine-tuned and for security we cannot visit the configuration pages from outside our own network.
All operations with strings are removed now. Should yield more- and more stable heap. The info- and log pages got a more decent look now. All the improvements from the ES8266 version are copied over to this project. 

The latest edition: [ESP32-ECU-v0_6](https://github.com/patience4711/ESP32-read-APS-inverters/blob/main/ESP32-ECU_v0-6.bin)

**Comment by JanBoos** [comment]
Update naar 1b geeft bij mij een foutmelding:  [HTTP ERROR] Bad Request


![image](https://user-images.githubusercontent.com/47677403/233383444-d4d43254-eb6d-4546-ac56-aea9388105cf.png)

**Comment by patience4711** [comment]
@JanBoos Can you tell what exactly goes wrong, is this after uploading the file? (after uploading you should reboot)

I flash all my tests and experiments via ota, must be a hundred times and it always works. Could you try it again with this version?

**Comment by JanBoos** [comment]
With 1c I get the same error (also with a different wifi-router). I also tried the ota with an esp8266 (9.8b) and that works ok. I read on the elegantota forum that more esp-32's have this problem. Could it be the partition-table.bin used?

Furthermore, the software on the esp32 works a lot smoother!!!

Unfortunately, my 3 APS DS3 inverters will not arrive until the beginning of May.

**Comment by patience4711** [comment]
@JanBoos 
 I managed to get rid of that annoying elegantOta and installed my own updatemanager. 
For me this works fine. I hope that your problem is solved with it.<br>

[ version 1d](https://github.com/patience4711/ESP32-read-APS-inverters/blob/main/ESP32-ECU_v0-1d.bin)<br>

**Comment by JanBoos** [comment]
It functions!! I first flashed 1d serially and then updated 1d again with the new OTA update. 

Thank you. Now I don't have to take my ECU + zigbee apart again and again :).

**Comment by patience4711** [comment]
Yes that's the idea of OTA.  
B.t.w.  _the software on the esp32 works a lot smoother!!!_
What do you mean by that? Smoother than ?

**Comment by JanBoos** [comment]
This means that the ECU software on the ESP32 responds much faster than on the ESP8266

**Comment by patience4711** [comment]
The latest state of the art is [v0_2](https://github.com/patience4711/ESP32-read-APS-inverters/blob/main/ESP32-ECU_v0-2.bin)

I improved the details webpage and solved some problems with crashes during polling.

**Comment by patience4711** [comment]
I tinkered a bit on the engine, resulting in a more efficient program. Moreover a re-installed the console as it is very handy if you don't have a computer nearby the ESP32-ECU

[version 0-3](https://github.com/patience4711/ESP32-read-APS-inverters/blob/main/ESP32-ECU_v0-3.bin)

If you update your existing ESP32-ECU with this version, please put off the "auto polling" (settings) first.

**Comment by patience4711** [comment]
I am now testing with version [ESP32-ECU-v0_6](https://github.com/patience4711/ESP32-read-APS-inverters/blob/main/ESP32-ECU_v0-6.bin)
I introduced a new way of communication between server and webpage. Instead of frequently unnecessary requests to the server and page reloads, the server notifies the page when new data becomes available, or something has changed in the state (zigbee down, sleepmodus, etc).  The webpage only makes a request than. This reduces the wifi trafic and the cpu dramatically.  Also there is no simultaneous polling and data request handling anymore, which reduces the chances of memory erosion.

**Comment by EntryGoon** [comment]
Found here that using ESP-Flasher (based on Pyflasher) is the only way to get the ESP32 flashed. unfortunately, I've spent the last hour, to found the web page credential without success... the password is "0000" by default, but the login is not defined anywhere and I can not log in the web page... And I'm pretty sure I just missed the information somewhere... ? thanks a lot for that great work !!!

**Comment by YellowSub** [comment]
Bonjour,

I'm using your software in ESP-ECU-v7_01 with Wemos D1 for 3 YC600 inverters since more than one year wihout issue, but now I have 3 other inverters DS3-L;  I've upgraded to V10_8 but my DS3-L are paired but not polled (my 3 YC600 are always ok).
I' using a CC2531 . Shall I change the software of the CC2531 ( actual is CC2530ZNP-with-SBL.hex) ?  I saw somewhere a  CC2531-DS3 version. It's not possible to poll YC600 and DS3 together?

Thank you very much for sharing...
